### ISSUE TITLE
Bun SQLite implementation support

### ISSUE THREAD

**Issue by Linux123123** [eslachance/enmap#120]
With the recent release of Bun v1.0, more people might be using bun over node. Bun has a native implementation of SQLite, which they claim is 3-6x faster than `better-sqlite-3`.

The API of the native module is inspired by `better-sqlite-3`, so it might be very easy to add support for it too.
*(Haven't looked at exactly what is used in `enmap` and how the actually easy it would be to add it)*

Bun support shouldn't interfere with existing node users and should only complement anyone using bun.
*(It should check if the process is running on bun and use the native implementation and there should also be an option to force it to run on `better-sqlite-3`.)*

**Comment by eslachance** [comment]
No work will be done on Bun until which time they are **actually** a drop-in replacement for Node. They are currently not. Not only is bun not available fully on Windows, it is also not actually feature-complete, as evident by its need for node to be installed for some features to be used. 

If Bun was a drop-in replacement, it should not require any effort whatsoever on the part of any developer of modules for it to be taken advantage of. If their native sqlite implementation is to be the greatest one, they should be doing this detection automatically with no effort on my part. 

TL;DR: I do not have any plans currently to make any effort to "have to support" any alternate runtime, whether it's bun or Deno.

**Comment by Linux123123** [comment]
Fair enough, thanks for looking into this.
